src/tool/hpcrun/memory/hpcrun-malloc.h

```diff
@@ -73,7 +74,7 @@
 extern "C" {
 #endif
 
-#if ! defined(VALGRIND) || defined(_IN_MEM_C)
+#if 1 // ENABLE_VG_ANNOTATIONS != 1 || defined(_IN_MEM_C)
```

Comment:
this looks useless and perhaps should be omitted.

Reply:
By "this" do you mean the entire #if...#else block, or just this change? Enabling the "Valgrind-safe" allocation methods causes segfaults at runtime.

Reply:
The purpose of hpcrun_malloc is to support allocation in signal handlers. The purpose of hpcrun_malloc_safe is to support allocation outside signal handlers using our allocator in a way that won't be interrupted by signals.

The #else of the #if block simply rewrites these as malloc. That won't work in hpcrun. Is there any reason why the #if 1, the #else and the block that follows, and the #endif should be retained. If not, they should be discarded.

Reply:
Doesn't seem to be a reason. Removed.